src/meili/map_matcher.cc

```diff
@@ -709,6 +711,11 @@ std::vector<MatchResults> MapMatcher::OfflineMatch(const std::vector<Measurement
 
   // Separate the measurements we are using for matching from the ones we'll just interpolate
   auto interpolated = AppendMeasurements(measurements);
+  // Should no matching edge candidates be found for the minimum required number of points, throw a
+  // 443 - NoSegment error code.
+  if (interpolated.size() < 2) {
```

Comment:
sadly this wont work. interpolated is just those candidates that were deemed appropriate for interpolation. you'll need to extend the interface of AppendMeasurements to return the total number of candidates that were found and if this number is 0 then we throw as you have done.

Reply:
Yeah, that totally makes sense re: `interpolated`.

`StateContainer` though, which is accessible in `MapMatcher` does contain a candidate list for each measurement state ID. Instead of changing the `AppendMeasurements` interface, we could use `StateContainer` to check if there are sufficient candidates from the search?

Reply:
k, pushing an example in just a sec.